internal/services/firewall/firewall_policy_application_rule_collection_resource.go

```diff
@@ -0,0 +1,432 @@
+								ValidateFunc: validation.StringIsNotEmpty,
+							},
+						},
+						"terminate_tls": {
+							Type:     pluginsdk.TypeBool,
+							Optional: true,
+						},
+						"web_categories": {
+							Type:     pluginsdk.TypeList,
+							Optional: true,
+							Elem: &pluginsdk.Schema{
+								Type:         pluginsdk.TypeString,
+								ValidateFunc: validation.StringIsNotEmpty,
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceFirewallPolicyApplicationRuleCollectionCreate(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Firewall.FirewallPolicyRuleGroupClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+	ruleCollectionGroupId, err := parse.FirewallPolicyRuleCollectionGroupID(d.Get("rule_collection_group_id").(string))
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, ruleCollectionGroupId.ResourceGroup, ruleCollectionGroupId.FirewallPolicyName, ruleCollectionGroupId.RuleCollectionGroupName)
+	if err != nil {
+		if !utils.ResponseWasNotFound(resp.Response) {
+			return fmt.Errorf("checking for existing Firewall Policy Rule Collection Group %q (Resource Group %q / Policy %q): %+v", ruleCollectionGroupId.RuleCollectionGroupName, ruleCollectionGroupId.ResourceGroup, ruleCollectionGroupId.FirewallPolicyName, err)
+		}
+	}
+
+	idx := indexFunc(*resp.RuleCollections, func(ruleCollection network.BasicFirewallPolicyRuleCollection) bool {
+		info, _ := ruleCollection.AsFirewallPolicyFilterRuleCollection()
+		return *info.Name == name
+	})
+
+	if resp.ID != nil && *resp.ID != "" && idx != -1 {
+		return tf.ImportAsExistsError("azurerm_firewall_policy_application_rule_collection", *resp.ID)
+	}
+
+	locks.ByName(ruleCollectionGroupId.FirewallPolicyName, AzureFirewallPolicyResourceName)
+	defer locks.UnlockByName(ruleCollectionGroupId.FirewallPolicyName, AzureFirewallPolicyResourceName)
+
+	rules := expandFirewallPolicyRuleApplication(d.Get("rule").([]interface{}))
+	ruleCollection := &network.FirewallPolicyFilterRuleCollection{
+		Action: &network.FirewallPolicyFilterRuleCollectionAction{
+			Type: network.FirewallPolicyFilterRuleCollectionActionType(d.Get("action").(string)),
+		},
+		Name:               utils.String(d.Get("name").(string)),
+		Priority:           utils.Int32(int32(d.Get("priority").(int))),
+		RuleCollectionType: network.RuleCollectionTypeFirewallPolicyFilterRuleCollection,
+		Rules:              rules,
+	}
+
+	if idx == -1 {
+		ruleCollections := append(*resp.RuleCollections, ruleCollection)
+		resp.RuleCollections = &ruleCollections
+	} else {
+		(*resp.RuleCollections)[idx] = ruleCollection
+	}
+
+	param := network.FirewallPolicyRuleCollectionGroup{
+		FirewallPolicyRuleCollectionGroupProperties: &network.FirewallPolicyRuleCollectionGroupProperties{
+			Priority:        resp.Priority,
+			RuleCollections: resp.RuleCollections,
+		},
+	}
```

Comment:
Can we simply re-assign the `RuleCollections` of the `resp` to avoid missing constructing any new properties in the future?